main.go

```diff
@@ -129,9 +135,45 @@ func wrappedMain() int {
 	// Make sure we clean up any managed plugins at the end of this
 	defer plugin.CleanupClients()
 
-	// Get the command line args. We shortcut "--version" and "-v" to
-	// just show the version.
+	// Get the command line args.
 	args := os.Args[1:]
+
+	// Prefix the args with any args from the EnvCLI
+	if v := os.Getenv(EnvCLI); v != "" {
+		log.Printf("[INFO] %s value: %q", EnvCLI, v)
+		extra, err := shellwords.Parse(v)
+		if err != nil {
+			Ui.Error(fmt.Sprintf(
+				"Error parsing extra CLI args from %s: %s",
+				EnvCLI, err))
+			return 1
+		}
+
+		// Find the index to place the flags. We put them exactly
+		// after the first non-flag arg.
+		idx := -1
+		for i, v := range args {
+			if v[0] != '-' {
```

Comment:
An arg can be 0 length -- even more likely in an automated system like TFE. 
`strings.HasPrefix`?